src/coredump/coredump.c

```diff
@@ -294,6 +308,33 @@ static int make_filename(const char *context[_CONTEXT_MAX], char **ret) {
         return 0;
 }
 
+static int open_coredump_tmpfile(const char *target, char **ret_filename) {
+        _cleanup_free_ char *tmp = NULL;
+        int fd;
+        int r;
+
+        assert(target);
+        assert(ret_filename);
+
+        fd = open("/var/lib/systemd/coredump", O_TMPFILE|O_CLOEXEC|O_NOCTTY|O_RDWR, 0640);
+        if (fd < 0) {
+                log_debug_errno(errno, "Failed to use O_TMPFILE: %m");
+
+                r = tempfn_random(target, NULL, &tmp);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to determine temporary file name: %m");
+
+                fd = open(tmp, O_CREAT|O_EXCL|O_RDWR|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0640);
+                if (fd < 0)
+                        return log_error_errno(errno, "Failed to create coredump file %s: %m", tmp);
```

Comment:
Hmm. Indeed it seems this behavior (or a related one via a new system call) is explicitly forbidden for security reasons 
http://marc.info/?l=linux-kernel&m=104973707408577
http://marc.info/?l=linux-kernel&m=104965552118236&w=2

So the solution was to add a parameter to `open` ... which turned out to be `O_TMPFILE`. OK, sorry for the noise.